xbmc/video/VideoDatabase.cpp

```diff
@@ -271,6 +274,10 @@ void CVideoDatabase::CreateAnalytics()
 
   m_pDS->exec("CREATE INDEX ix_rating ON rating(media_id, media_type(20))");
 
+  m_pDS->exec("CREATE INDEX ix_uniqueid1 ON uniqueid(uniqueid_id, media_id, media_type(20))");
+  m_pDS->exec("CREATE INDEX ix_uniqueid2 ON uniqueid(uniqueid_id, media_id, value(20))");
+  m_pDS->exec("CREATE INDEX ix_uniqueid3 ON uniqueid(media_type(20), value(20))");
```

Comment:
You're `WHERE` clauses filtering on `uniqueid` use the following combinations:
* `media_id` and `media_type`
* `media_id`, `media_type` and `type`
* `media_type` and `value` (only as part of `SetDetailsForFoo()`)
* `uniqueid_id` (should already be covered by `PRIMARY KEY`)
so you should create indices for at least the first two (without `uniqueid_id`).